faq.html

```diff
@@ -161,6 +161,87 @@ <h2 id="outputformat">The output format of ParticleFile</h2>
 
   <hr>
+  <h2 id="performance">Memory Behaviour and Performance</h2>
+
+  Parcels, with its Lagrangian simulation approach and the commonly large amount of particles being simulated, is a computationally-demanding application. Supported models, such as CMEMS and NEMO, add to the complexity as they require large blocks of memory to be available. Despite the continuous improvements in performance and efficiency, it is thus possible to encounter performance- and memory-related problems. The following paragraphs are meant as a guideline to resolve those related issues.
+  <p></p>
+  The first step in tracking memory-related issues is a reasonability analysis: can issues that you are possibly experiencing actual stem from memory exhaustion ? Get an overview of the models and fields that you are using, consider if you're running the application with- or without depth information, and keep in mind that at each time in the simulation you max. need space for 3 timestamps of all fields in addition to the actual particles. Does this really exhaust the memory ? As a safeguard check, you can run a subset of your simulation (e.g the first day or week) locally while tracking the memory consumption with tools such as the <b>Task Manager</b> (Windows), the <b>Activity Monitor</b> (MacOS) or the <b>System Monitor</b> (Linux).
+  <p></p>
+  Suppose that you have determined that reason for your simulation not finishing <i>can be</i> or <i>is</i> memory exhaustion. Then, the next step is to determine if the cause of memory exhaustion is due to <b>Particle[Sets]</b> or <b>Field[Sets]</b>. Fortunately, this is easy to assess:
+  <ul>
+    <li>Initialize your ParcileSet with only few particles, e.g. <code class="language-python">pset = ParticleSet(fieldset=fieldset, pclass=JITParticle, lon=np.random.rand(16,1), lat=np.random.rand(16,1))</code></li>
+    <li>Do not re-add particles periodically via <code class="language-python">repeatdt</code>, e.g. change <code class="language-python">pset = ParticleSet(..., lat=np.random.rand(16,1), repeatdt=timedelta(hours=1))</code> to <code class="language-python">pset = ParticleSet(..., lat=np.random.rand(16,1))</code></li>
+  </ul>
+  Run your simulation again and observe the memory with the tools mentioned above. If your simulation runs through without problems, it is almost certain your real simulation requires too many particles. There is no generic solution at this point for it, so: Get in touch with us at <a href="https://github.com/OceanParcels/parcels/issues">the OceanParcels GitHub page</a> via the issue tracker to find an individual solution.
```

Comment:
Is this really true? I would say that a generic solution if you have too many particles is: 'split your experiment up into multiple smaller runs'

Reply:
yes, that's a good alternative suggestion - shall I integrate that in text ?

When it comes to 'is this really true?': Okay, I should mention perhaps beforehand that the user should, in case of failure, try to reproduce the error with a minimal-complexity example. So: simple Particle[JIT]'s, simple advection kernel - be sure that calculation errors are ruled out. Possibly I should start with that pre-condition in this passage.

After that, it is actually true; if the user runs any simulation with just 16 JIT Particles on a field without runtime limit (so: 1 temporally-complete simulation) and it works, one can be sure the error must be in the particles. After all, that's what we/you usually suggest testing in common trouble-shooting issues, right ?

Reply:
Yes, very good point. Please go ahead and integrate into the text?